src/views/ModelPlan/Documents/AddDocument/LinkDocument.tsx

```diff
@@ -0,0 +1,416 @@
+                        key={`Error.${key}`}
+                        errorKey={key}
+                        message={flatErrors[key]}
+                      />
+                    );
+                  })}
+                </ErrorAlert>
+              )}
+              <div>
+                <Form
+                  onSubmit={e => {
+                    handleSubmit(e);
+                    window.scrollTo(0, 0);
+                  }}
+                >
+                  <FieldGroup scrollElement="url" error={!!flatErrors.url}>
+                    <Label htmlFor="FileUpload-LinkDocument">
+                      {t('linkDocument.linkLabel')}
+                      <RequiredAsterisk />
+                    </Label>
+
+                    <Label
+                      htmlFor="FileUpload-LinkDocument"
+                      hint
+                      className="text-normal text-base margin-y-1"
+                    >
+                      {t('linkDocument.linkHelpText')}
+                    </Label>
+
+                    <FieldErrorMsg>{flatErrors.url}</FieldErrorMsg>
+
+                    <Field
+                      as={TextInput}
+                      id="FileUpload-LinkDocument"
+                      name="url"
+                    />
+                  </FieldGroup>
+                  <FieldGroup scrollElement="name" error={!!flatErrors.name}>
+                    <Label htmlFor="FileUpload-LinkFileName">
+                      {t('linkDocument.fileNameLabel')}
+                      <RequiredAsterisk />
+                    </Label>
+
+                    <Label
+                      htmlFor="FileUpload-LinkFileName"
+                      hint
+                      className="text-normal text-base margin-y-1"
+                    >
+                      {t('linkDocument.fileNameHelpText')}
+                    </Label>
+
+                    <FieldErrorMsg>{flatErrors.name}</FieldErrorMsg>
+
+                    <Field
+                      as={TextInput}
+                      id="FileUpload-LinkFileName"
+                      name="name"
+                    />
+                  </FieldGroup>
+
+                  <FieldGroup
+                    id="file-type"
+                    scrollElement="documentType"
+                    error={!!flatErrors.documentType}
+                  >
+                    <fieldset className="usa-fieldset margin-top-4">
+                      <legend className="usa-label">
+                        {t('whatType')}
+                        <RequiredAsterisk />
+                      </legend>
+                      <FieldErrorMsg>{flatErrors.documentType}</FieldErrorMsg>
+                      {(Object.keys(DocumentType) as Array<
+                        keyof typeof DocumentType
+                      >)
+                        .filter(documentType => documentType !== 'OTHER') // Filter 'OTHER' to be last in the radio button list
+                        .map(documentType => {
```

Comment:
We have the util `getKeys` that works for this implementation of iterationg over typescript enums that preserves type.  We also have the `sortOtherEnum` util that will automatically sort `Other` enums as last in order.